### PR TITLE
Restore default role check in `make check`.

### PR DESCRIPTION
**Pull request by ezio-melotti**
This PR restores the check for the default role (see GH-92289) by doing two things. First, it sets the `severity` to `0` adding `--severity=0`, so that failing the default role check results in an error.  This however also enables the long lines check, so it explicitly disabled it with `--disable='line too long'`.  There are currently no other checks with `severity=0`.

This PR only fixes `Makefile`.  If this approach is sound, `make.bat` should be fixed too.

Using `--enable` instead might be a better approach, once sphinx-contrib/sphinx-lint#27 lands.

As a side note, it took me a bit to figure out that `--disable=` expected a message rather than the name of a checker.

cc @JulienPalard

### CODE REVIEW

**Comment by ezio-melotti** [comment]
With this PR, [the check](https://dev.azure.com/Python/cpython/_build/results?buildId=102551&view=logs&j=4db1505a-29e5-5cc0-240b-53a8a2681f75&t=a975920c-8356-5388-147c-613d5fab0171) successfully failed:
```sh
PATH=./venv/bin:$PATH sphinx-lint -i tools -i ./venv -i README.rst --severity=0 --disable='line too long'
No problems found.
PATH=./venv/bin:$PATH sphinx-lint ../Misc/NEWS.d/next/ --severity=0 --disable='line too long'
[0] ../Misc/NEWS.d/next/Library/2022-04-26-18-02-44.gh-issue-91928.V0YveU.rst:1: default role used (hint: for inline code, use double backticks)
[0] ../Misc/NEWS.d/next/Documentation/2022-04-24-22-09-31.gh-issue-91888.kTjJLx.rst:1: default role used (hint: for inline code, use double backticks)
2 problems with severity 0 found.
make: *** [Makefile:217: check] Error 1
##[error]Bash exited with code '2'.
```

File: Doc/Makefile
Comment (vstinner) [inline Doc/Makefile:217]: Would you mind to add a comment to explain the purpose of the --severity option? Summarize what you wrote in the PR description ;-)

Comment (ezio-melotti) [reply Doc/Makefile:217]: Done.

**Comment by ezio-melotti** [comment]
I added the same changes to `make.bat`, but I don't have a Windows machine to test them (cc @python/windows-team).

`make.bat` seems outdated compared to `Makefile`, so I made a couple of changes to make them converge:
* https://github.com/python/cpython/commit/b9735420aa8ecd2555fe3dc000863bc50487334b added the `sphinx-lint` check to `Misc/NEWS.d/next` (see https://github.com/python/cpython/issues/86404#issuecomment-1093890004)
* https://github.com/python/cpython/commit/b3f1f59cf451d4a25b204e7a24f7be4c95e40be8 added `README.rst` to the ignored files, but I don't see a reason to ignore it, so I removed it from `Makefile` to align it with `make.bat` instead (see 
https://github.com/python/cpython/pull/104#issuecomment-279921094)

In addition, https://github.com/python/cpython/commit/bfba2cd40629e52ba74ddc75a3415adc8f0ef71b added the `venv` dir to the ignored dirs, but `make.bat` doesn't seem to use venv, so no change was required.

**Comment by vstinner** [comment]
The CI fails:

```

Error: [0] ../Misc/NEWS.d/next/Documentation/2022-04-24-22-09-31.gh-issue-91888.kTjJLx.rst:1: default role used (hint: for inline code, use double backticks)
Error: [0] ../Misc/NEWS.d/next/Library/2022-04-26-18-02-44.gh-issue-91928.V0YveU.rst:1: default role used (hint: for inline code, use double backticks)
```

**Comment by ezio-melotti** [comment]
Yes, I'm leaving the failures so that people can test if they are detected on Windows too.  If they are, I'll fix them and merge the PR.

**Comment by AlexWaygood** [comment]
Tried it out on my Windows machine:

```
C:\Users\alexw\coding\cpython>"Doc/make.bat" check
Using py -3.10 (found 3.10 with py.exe)
Error: path too does not exist
Error: path too does not exist
```

(I'll be honest, I usually don't bother with the `.bat` file when I'm writing a docs patch. I usually use `sphinx-build` to build the docs locally, `git diff` to check for trailing whitespace, and `sphinx-lint` in CPython's CI to check for markup errors.)

File: Doc/make.bat
Comment (AlexWaygood) [inline Doc/make.bat]: If I make these changes then the `.bat` script picks up the two true-positive lint errors in exactly the same way as the makefile

```suggestion
cmd /S /C "%SPHINXLINT% -i tools --severity=0 --disable=^"line too long^""
cmd /S /C "%SPHINXLINT% --severity=0 --disable=^"line too long^" ..\Misc\NEWS.d\next\ "
```

Output I get with these changes applied:

```
C:\Users\alexw\coding\cpython>"Doc/make.bat" check
Using py -3.10 (found 3.10 with py.exe)
No problems found.
[0] ..\Misc\NEWS.d\next\Documentation\2022-04-24-22-09-31.gh-issue-91888.kTjJLx.rst:1: default role used (hint: for inline code, use double backticks)
[0] ..\Misc\NEWS.d\next\Library\2022-04-26-18-02-44.gh-issue-91928.V0YveU.rst:1: default role used (hint: for inline code, use double backticks)
2 problems with severity 0 found.
```

Comment (ezio-melotti) [reply Doc/make.bat]: Thanks for testing and for the fix!

Apparently there are [a few ways of escaping quotes in bat files](https://stackoverflow.com/a/31413730): `^"`, `\"`, `""`, and `"""`.  If `\"`/`""` also work, I have a slight preference for those, since they are more commonly used and understood than `^"`.

Comment (AlexWaygood) [reply Doc/make.bat]: Hmm, from manual experimentation on my machine, only `^"` seems to work here. (I'd also rather go with a backslash or double-quote if possible! But it doesn't seem to be possible here 😕)

Comment (zware) [reply Doc/make.bat]: Agreed with Alex, `^"` is unfortunately the way to go.  Except when it isn't, but here it probably is.

Escaping in `.bat` is a nightmare.

**Comment by ezio-melotti** [comment]
Thanks everyone for the feedback and testing!  I'll wait for the feedback of @JulienPalard before fixing the 2 failures and merging the PR.

**Comment by JulienPalard** [comment]
@ezio-melotti Don't you prefer waiting for https://github.com/sphinx-contrib/sphinx-lint/pull/27 to be merged, avoiding two commits on cpython doing the same thing?

It would look like `sphinx-lint --enable default-role`, which is more transparent than `--severity 0 --disable 'line too long'`.

File: Doc/Makefile
Comment (JulienPalard) [inline Doc/Makefile]:
```suggestion
	$(SPHINXLINT) -i tools -i $(VENVDIR) --enable default-role
```

Finally :)

I appreciate that the PR titled "Restore default role check" contains `--enable default-role` instead of `--disable line too long` :)

**Comment by ezio-melotti** [comment]
The PR should be ready to be merged.
@AlexWaygood / @zware: can you check if it works on Windows? (you might have to update `sphinx-lint`)

**Comment by ezio-melotti** [comment]
Looks like [it works on WIndows](https://dev.azure.com/Python/cpython/_build/results?buildId=103618&view=logs&j=4db1505a-29e5-5cc0-240b-53a8a2681f75&t=a975920c-8356-5388-147c-613d5fab0171&l=14) too.

**Review by pganssle — APPROVED** [review]

**Comment by JulienPalard** [comment]
Thanks @ezio-melotti!

**Pull request merged**
